Build Fail: [Kerror: [Ksubuid_count undeclared (first use in this function)

As of commit (https://github.com/shadow-maint/shadow/commit/c4539fc4f9e23874ce1cdd7265fb28d5a44e49eb), build is borked on my Gentoo machine using GCC 11.2 version. Here is a snippet of the error, and the build options used. full log attached.

https://github.com/shoober420/shoober420-overlay/blob/main/sys-apps/shadow/shadow-9999.ebuild#L61

[shadowbuild.log](https://github.com/shadow-maint/shadow/files/6987295/shadowbuild.log)

```
/bin/sh ../libtool  --tag=CC   --mode=link x86_64-pc-linux-gnu-gcc  -march=native -Ofast -pipe -fno-plt -fno-common -fno-semantic-interposition -fipa-pta -falign-functions=32 -fdevirtualize-at-ltrans -fuse-linker-plugin -floop-nest-optimize -fgraphite-identity -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -fno-math-errno -ffinite-math-only -fno-rounding-math -fno-signaling-nans -fcx-limited-range -fexcess-precision=fast -w -flto=auto  -Wl,-O1 -Wl,--as-needed -o expiry expiry.o  ../libmisc/libmisc.la ../lib/libshadow.la   
/bin/sh ../libtool  --tag=CC   --mode=link x86_64-pc-linux-gnu-gcc  -march=native -Ofast -pipe -fno-plt -fno-common -fno-semantic-interposition -fipa-pta -falign-functions=32 -fdevirtualize-at-ltrans -fuse-linker-plugin -floop-nest-optimize -fgraphite-identity -fno-signed-zeros -fno-trapping-math -fassociative-math -freciprocal-math -fno-math-errno -ffinite-math-only -fno-rounding-math -fno-signaling-nans -fcx-limited-range -fexcess-precision=fast -w -flto=auto  -Wl,-O1 -Wl,--as-needed -o gpasswd gpasswd.o  ../libmisc/libmisc.la ../lib/libshadow.la    -lcrypt  
[01m[Kuseradd.c:[m[K In function '[01m[Kmain[m[K':
[01m[Kuseradd.c:2643:21:[m[K [01;31m[Kerror: [m[K'[01m[Ksubuid_count[m[K' undeclared (first use in this function)
 2643 |         usr_update ([01;31m[Ksubuid_count[m[K, subgid_count);
      |                     [01;31m[K^~~~~~~~~~~~[m[K
[01m[Kuseradd.c:2643:21:[m[K [01;36m[Knote: [m[Keach undeclared identifier is reported only once for each function it appears in
[01m[Kuseradd.c:2643:35:[m[K [01;31m[Kerror: [m[K'[01m[Ksubgid_count[m[K' undeclared (first use in this function)
 2643 |         usr_update (subuid_count, [01;31m[Ksubgid_count[m[K);
      |                                   [01;31m[K^~~~~~~~~~~~[m[K
make[2]: *** [Makefile:1226: useradd.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

--- a/src/useradd.c
+++ b/src/useradd.c
@@ -2418,9 +2418,9 @@ int main (int argc, char **argv)
 #ifdef ENABLE_SUBIDS
 	uid_t uid_min;
 	uid_t uid_max;
-	unsigned long subuid_count;
-	unsigned long subgid_count;
 #endif
+	unsigned long subuid_count = 0;
+	unsigned long subgid_count = 0;
 
 	/*
 	 * Get my name so that I can use it to report errors.

